subprojects/docs/src/docs/userguide/customTasks.adoc

```diff
@@ -250,6 +250,108 @@ Using Gradle's `IncrementalTaskInputs` is not the only way to create tasks that
 
 However, if the state files are non-relocatable, then they can't be shared via the build cache. Indeed, when the task is loaded from cache, any such state files must be cleaned up to prevent stale state to confuse the tool during the next execution. Gradle can ensure such stale files are removed if they are declared via `task.localState.register()` or a property is marked with the `@LocalState` annotation.
 
+[[sec:declaring_and_using_command_line_options]]
+=== Declaring and Using Command Line Options
```

Comment:
we should link to this section from the command line chapter, I think, adding some text to that chapter to say that this is possible via plugins.

Reply:
done